### DISCUSSION TITLE
Cleanup build account in resulting golden images

### DISCUSSION THREAD

**Discussion by ukushisanet** [vmware/packer-examples-for-vsphere#271] [Help] [answered]
Hi Ryan,
Thank you very much for your work, it's a very useful project ⭐.

I wanted to know if there was an easy way to clean up all references to the "build user" account at the end of the image build process?

The only way to do that I found is to use the root account and secure the sshd_config file at the end of the process otherwise impossible to clean a (currently connected) user account without cutting the connection of the ssh communicator of packer.

Do you have a better idea?

**Comment by tenthirtyam** [comment, marked as answer]
Thanks for the ⭐ _'ing_ the project.

Unfortunately, I've not considered this _(yet. - good potential enhancement)_, but I imagine that this is a similar issue amoungst many other plugins and their builders.

Tagging some pals @nywilken and @kmruddy to see if they have some ideas that may assist.

Perhaps a first-boot script?

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Reply by tenthirtyam** [reply]
Or perhaps, better yet, including the process in the `shutdown_command`:

```hcl
shutdown_command   = "sudo su root -c \"userdel -rf rainpole; /sbin/shutdown -hP now\""
```

Haven't tested this yet.

Ryan

**Reply by ukushisanet** [reply]
Hi Ryan,

Thanks for this quick answer.

You're right, that's more a general topic with Packer, and thanks again to take the time to answer this. I haven't looked in the Packer issues yet to see if this question has been asked before but I will because it must be.

The first-boot script solution was also mine but does not suit me too much because it means putting even more things in the golden image that I want to keep as close to a minimal install as possible.

And for the second one, I can tell you that it works, I tested it, I may be reviewing the use of the root account for the benefit of this solution. The "shutdown_command" variable doesn't really have the same meaning with this anymore... but you just have to know that.

Regards,
